caikit_tgis_backend/tgis_backend.py

```diff
@@ -164,6 +179,24 @@ def stop(self):
             log.debug("Unloading model %s on stop", model_id)
             self.unload_model(model_id)
 
+    def handle_runtime_context(
+        self,
+        model_id: str,
+        runtime_context: RuntimeServerContextType,
+    ):
+        """Handle the runtime context for a request for the given model"""
+        if route_info := self.get_route_info(runtime_context):
+            log.debug(
+                "<TGB10705560D> Registering remote model connection with context "
+                "override: 'hostname: %s'",
+                route_info,
+            )
+            self.register_model_connection(
```

Comment:
Currently we are not "registering" the connection if the `model_id` is present in the `self._model_connection`.. I was thinking what if the route info has changed for some reason between calls.

Should we check if the context / route-info is same as the previously configured `route-info` in the `register_model_connection` function and then update the connection / initiate new connection, in case the value has changed? This will allow dynamically refreshing the connection automatically ? This will have a bit of overhead, but if the client is requesting a new route, then this would get handled ?

Reply:
@mynhardtburger and I thought about this with the initial implementation, but decided against it for simplicity. What's written here preserves that decision, though I think relaxing it in the future would be fine.